core/arch/riscv/kernel/spinlock.c

```diff
@@ -0,0 +1,29 @@
+// SPDX-License-Identifier: BSD-2-Clause
+/*
+ * Copyright 2022 NXP
+ */
+
+#include <kernel/spinlock.h>
+
+void __cpu_spin_lock(unsigned int *lock)
+{
+	while (__cpu_spin_trylock(lock))
+		;
+}
+
+void __cpu_spin_unlock(unsigned int *lock)
+{
+	asm volatile ("fence rw,w" ::: "memory");
+	asm volatile ("amoswap.w x0, x0, %0" : "=A" (*lock) :: "memory");
```

Comment:
We try to avoid inline assembly instead we code the entire function in an assembly file or use some helper wrapper functions.

In this case I suppose it would be better to just implement it in an assembly file. Same for `__cpu_spin_trylock()` below.